components/dashboard/src/components/UsageBasedBillingConfig.tsx

```diff
@@ -180,55 +200,201 @@ export default function UsageBasedBillingConfig({ subject, attributionId }: Prop
+                        </div>
+                        <div className="bg-gray-100 dark:bg-gray-800 border-t border-gray-200 dark:border-gray-800 -m-4 p-4 mt-8 rounded-b-xl">
+                            <div className="uppercase text-sm text-gray-400 dark:text-gray-500">Upgrade Plan</div>
+                            <div className="mt-1 text-xl font-semibold flex-grow text-gray-500 dark:text-gray-400">
+                                {currency === "EUR" ? "€" : "$"}9 / month
+                            </div>
+                            <div className="mt-4 flex space-x-1 text-gray-400 dark:text-gray-500">
+                                <Check className="m-0.5 w-5 h-5" />
+                                <div className="flex flex-col">
+                                    <span className="font-bold">1,000 credits</span>
+                                </div>
+                            </div>
+                            <div className="mt-2 flex space-x-1 text-gray-400 dark:text-gray-500">
+                                <Check className="m-0.5 w-5 h-5" />
+                                <div className="flex flex-col">
+                                    <span className="font-bold">Pay-as-you-go after 1,000 credits</span>
+                                    <span>36¢ for 10 credits or 1 hour of Standard workspace usage.</span>
+                                </div>
+                            </div>
+                            <div className="mt-5 flex flex-col">
+                                <button className="self-end" onClick={() => setShowBillingSetupModal(true)}>
+                                    Upgrade Plan
+                                </button>
+                            </div>
+                        </div>
+                    </div>
+                )}
                 {showManageBilling && (
                     <div className="max-w-xl flex space-x-4">
-                        <div className="flex flex-col w-72 mt-4 h-32 p-4 rounded-xl bg-gray-100 dark:bg-gray-800">
-                            <div className="uppercase text-sm text-gray-400 dark:text-gray-500">Billing</div>
-                            <div className="text-xl font-semibold flex-grow text-gray-600 dark:text-gray-400">
-                                Active
-                            </div>
-                            <a className="self-end" href={stripePortalUrl}>
+                        <div className="flex-grow flex flex-col mt-4 p-4 rounded-xl bg-gray-50 dark:bg-gray-900">
+                            <div className="uppercase text-sm text-gray-400 dark:text-gray-500">Current Plan</div>
+                            {AttributionId.parse(attributionId)?.kind === "user" ? (
+                                <>
+                                    <div className="mt-1 text-xl font-semibold flex-grow text-gray-800 dark:text-gray-100">
+                                        {currency === "EUR" ? "€" : "$"}9 / month
```

Comment:
In preview environments, everyone always sees $ as the default. In production, the currency is pre-selected based on your region. This auto-select is just not configured for preview environments or staging.

Reply:
Ah, I see. Thanks for the insight! 💡 

Do you think this is trivial or worth the effort to replicate this in preview environments?

Reply:
Definitely not trivial, so probably not worth the effort. A cheap fix could be to make preview always show € instead of $ 🤷 but maybe that's also not really worth the effort. Please feel free to open an issue if you'd like this to be fixed. 😇